instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/patches/client.rb

```diff
@@ -51,18 +51,16 @@ module Client
           ].freeze
 
           def query(sql, options = {})
-            attributes = client_attributes
-            case config[:db_statement]
-            when :include
-              attributes['db.statement'] = sql
-            when :obfuscate
-              attributes['db.statement'] = obfuscate_sql(sql)
-            end
-            tracer.in_span(
-              database_span_name(sql),
-              attributes: attributes.merge!(OpenTelemetry::Instrumentation::Mysql2.attributes),
-              kind: :client
-            ) do
+            overrides = OpenTelemetry::Instrumentation::Mysql2.attributes
+            statement = config[:db_statement]
+            statement = :override if overrides.key?('db.statement')
+
+            attributes = client_attributes.merge!(overrides)
+            attributes['db.statement'] = sql if statement == :include
+
+            tracer.in_span(database_span_name(sql), attributes: attributes, kind: :client) do |span|
+              span['db.statement'] = obfuscate_sql(sql) if OpenTelemetry::Trace.current_span.recording? && statement == :obfuscate
```

Comment:
```suggestion
              span['db.statement'] = obfuscate_sql(sql) if statement == :obfuscate && OpenTelemetry::Trace.current_span.recording?
```

> Checking if the current span is recording also entails doing some computation, though, ...

If the `recording?` check is expensive, maybe switch the order of this guard to early return false when obfuscation is not enabled?

>  ... so we ought to benchmark this.

Was this ever benchmarked?